src/aleph_client/commands/utils.py

```diff
@@ -200,3 +206,111 @@ def is_environment_interactive() -> bool:
+            await asyncio.gather(fetch_task, update_task)
+
+    return item_hashes
+
+async def fetch_data(session: aiohttp.ClientSession, node_info: NodeInfo, queue: asyncio.Queue, progress: ProgressTable, task: list, item_hashes: list):
+    tasks = [fetch_and_queue(session, node, queue, progress, task, item_hashes) for node in node_info.nodes]
+    await asyncio.gather(*tasks)
+    await queue.put(None)
+
+async def fetch_and_queue(session: aiohttp.ClientSession, node: NodeInfo, queue: asyncio.Queue, progress: Progress, task: list, item_hashes: list):
+    try:
+        system_info, version = await asyncio.gather(
+            fetch_crn_system(session, node),
+            get_crn_version(session, node)
+        )
+        async with session.get(node["address"] + "status/check/ipv6", timeout=10) as resp:
+            if resp.status == 200:
+                node_stream = node["stream_reward"]
+                if node_stream and system_info:
+                    node_name = _escape_and_normalize(node["name"])
+                    node_name = _remove_ansi_escape(node_name)
+                    node_address = node["address"]
+                    score = _format_score(node["score"])
+                    cpu = f"{system_info['cpu']['count']} {system_info['properties']['cpu']['architecture']}"
+                    hdd = f"{system_info['disk']['available_kB'] / 1024 / 1024:.2f} GB"
+                    ram = f"{system_info['mem']['available_kB'] / 1024 / 1024:.2f} GB"
+                    await queue.put((score, node_name, cpu, ram, hdd, version, node_stream, node_address))
+                    item_hashes.append(node_stream)
+    except Exception as e:
+        
+        pass
+    finally:
+        progress.update(task, advance=1)
+
+async def update_table(queue: asyncio.Queue, table: Table):
+    while True:
+        data = await queue.get()
+        if data is None:
+            break
+        table.add_row(*data)
+
+async def fetch_crn_system(session: aiohttp.ClientSession, node: NodeInfo):
+    try:
+        async with session.get(node["address"] + "about/usage/system", timeout=10) as resp:
+            if resp.status == 200:
+                data = await resp.json()
+    except Exception as e:
+        data = None
+    return data
+
+async def get_crn_version(session: aiohttp.ClientSession, node: NodeInfo, retries=3):
+    version = None
+    timeout = aiohttp.ClientTimeout(connect=20, total=40)
+    
+    try:
+        async with session.get(node['address'], timeout=timeout) as resp:
+            if resp.status == 200:
+                    html_content = await resp.text()
+                    soup = BeautifulSoup(html_content, 'html.parser')
+                    p_tags = soup.find_all('p')
+                    for p in p_tags:
+                        if 'Running version' in p.text:
+                            version = p.find('i').text
+                            return version
```

Comment:
You may want to use the last `server` header in the response instead.

See this code from the metrics: https://github.com/aleph-im/aleph-scoring/blob/2444ce57c6ad5a6693be29675b451ddff0b3dfc4/aleph_scoring/metrics/__init__.py#L172-L190

Reply:
That is a best way to do it indeed, i modified it.